src/index.ts

```diff
@@ -126,90 +260,73 @@ const plugin: JupyterFrontEndPlugin<void> = {
+              title: 'Share Notebook',
+              body: new ShareDialog(),
+              buttons: [Dialog.cancelButton(), Dialog.okButton()]
+            });
 
-          if (result.button.accept) {
-            const { notebookName, password } = result.value as IShareDialogData;
-
-            try {
-              // Show loading indicator
-              // TODO: this doesn't show up in the dialog properly, we could
-              // even remove it as loading doesn't take long at all
-              const loadingIndicator = document.createElement('div');
-              loadingIndicator.textContent = 'Sharing notebook...';
-              loadingIndicator.style.position = 'fixed';
-              loadingIndicator.style.bottom = '20px';
-              loadingIndicator.style.right = '20px';
-              loadingIndicator.style.padding = '10px';
-              loadingIndicator.style.backgroundColor = '#f0f0f0';
-              loadingIndicator.style.borderRadius = '5px';
-              loadingIndicator.style.zIndex = '1000';
-              document.body.appendChild(loadingIndicator);
-
-              await sharingService.authenticate();
-
-              let shareResponse;
-              if (isNewShare) {
-                shareResponse = await sharingService.share(notebookContent, password);
-              } else if (notebookId) {
-                shareResponse = await sharingService.update(notebookId, notebookContent, password);
-              }
+            if (result.button.accept) {
+              const { notebookName } = result.value as IShareDialogData;
+              const password = generatePassword();
 
-              if (shareResponse && shareResponse.notebook) {
-                // We need to update the metadata in the notebookContent first
-                // to do this, and we need to ensure that the metadata object exists
-                if (!notebookContent.metadata) {
-                  notebookContent.metadata = {};
-                }
+              try {
+                await sharingService.token;
+                const shareResponse = await sharingService.share(notebookContent, password);
+
+                if (shareResponse && shareResponse.notebook) {
+                  if (!notebookContent.metadata) {
+                    notebookContent.metadata = {};
+                  }
 
-                notebookContent.metadata.sharedId = shareResponse.notebook.id;
-                notebookContent.metadata.readableId = shareResponse.notebook.readable_id;
-                notebookContent.metadata.sharedName = notebookName;
-                notebookContent.metadata.isPasswordProtected = true;
+                  notebookContent.metadata.sharedId = shareResponse.notebook.id;
+                  notebookContent.metadata.readableId = shareResponse.notebook.readable_id;
+                  notebookContent.metadata.sharedName = notebookName;
+                  notebookContent.metadata.isPasswordProtected = true;
```

Comment:
I think this can just be

```ts
                  notebookContent.metadata = {
                    ...notebookContent.metadata,
                    sharedId: shareResponse.notebook.id,
                    readable_id: shareResponse.notebook.readable_id,
                    sharedName: notebookName,
                    isPasswordProtected: true
                  };
```

since destructuring an `undefined` returns `{}`.